subprojects/core/src/integTest/groovy/org/gradle/api/tasks/options/TaskOptionIntegrationTest.groovy

```diff
@@ -321,4 +321,121 @@ Options
         then:
         result.assertTaskExecuted(':myTask').assertOutputContains('Serial: 4321')
     }
+
+    def "options from interfaces with same method defined in class should use overridden value"() {
+        given:
+        buildFile << '''
+            interface MyInterface {
+              @Option(
+                option = 'serial',
+                description = 'Target the device with given serial'
+              )
+              @Optional
+              @Input
+              Property<String> getSerial()
+            }
+
+            abstract class MyTask extends DefaultTask implements MyInterface {
+              @Option(
+                option = 'serialNumber',
+                description = 'Target the device with given serial'
+              )
+              @Optional
+              @Input
+              abstract Property<String> getSerial()
+
+              @TaskAction
+              void action() {
+                println "Serial: ${serial.getOrElse('-')}"
+              }
+            }
+
+            tasks.register("myTask", MyTask.class)
+        '''
+
+        when:
+        succeeds('myTask', '--serial=1234', '--serialNumber=4321')
+
+        then:
+        result.assertTaskExecuted(':myTask').assertOutputContains('Serial: 4321')
+    }
+
+    def "options from interfaces with same method defined twice with same name should work"() {
+        given:
+        buildFile << '''
+            interface MyInterface {
+              @Option(
+                option = 'serial',
+                description = 'Target the device with given serial'
+              )
+              @Optional
+              @Input
+              Property<String> getSerial()
+            }
+
+            interface MyInterface1 {
+              @Option(
+                option = 'serial',
+                description = 'Target the device with given serial'
```

Comment:
Could you make these descriptions different and assert which one we see with `help --task myTask`?